modules/admission-webhook-types.adoc

```diff
@@ -0,0 +1,124 @@
+// Module included in the following assemblies:
+//
+// * architecture/admission-controllers.adoc
+
+[id="admission-webhook-types_{context}"]
+= Types of admission webhooks
+
+Cluster administrators can include _mutating admission webhooks_ or _validating admission webhooks_
+in the admission chain of the API server.
+
+[[mutating-admission-webhooks]]
+*Mutating admission webhooks* are invoked during the mutation phase of the admission process, which allows modification of the resource content before it is persisted. One example of a mutating admission webhook is the
+ifdef::openshift-enterprise,openshift-origin[]
+Pod Node Selector
+endif::openshift-enterprise,openshift-origin[]
+ifdef::openshift-online,openshift-dedicated[]
+Pod Node Selector
+endif::openshift-online,openshift-dedicated[]
+feature, which uses an annotation on a namespace to find a label selector and add it to the pod specification.
+
+[[mutating-admission-webhooks-sample]]
+.Sample mutating admission webhook configuration:
+
+[source,yaml]
+----
+apiVersion: admissionregistration.k8s.io/v1beta1
+  kind: MutatingWebhookConfiguration <1>
+  metadata:
+    name: <controller_name> <2>
+  webhooks:
+  - name: <webhook_name> <3>
+    clientConfig: <4>
+      service:
+        namespace:  <5>
+        name: <6>
+       path: <webhook_url> <7>
+      caBundle: <cert> <8>
+    rules: <9>
+    - operations: <10>
+      - <operation>
+      apiGroups:
+      - ""
+      apiVersions:
+      - "*"
+      resources:
+      - <resource>
+    failurePolicy: <policy> <11>
+----
+
+<1> Specifies a mutating admission webhook configuration.
+<2> The name for the admission webhook object.
+<3> The name of the webhook to call.
+<4> Information about how to connect to, trust, and send data to the webhook server.
+<5> The project where the front-end service is created.
+<6> The name of the front-end service.
+<7> The webhook URL used for admission requests.
+<8> A PEM-encoded CA certificate that signs the server certificate used by the webhook server.
+<9> Rules that define when the API server should use this controller.
+<10> The operation(s) that triggers the API server to call this controller:
+* create
+* update
+* delete
+* connect
+<11> Specifies how the policy should proceed if the webhook admission server is unavailable. Either `Ignore` (allow/fail open) or `Fail` (block/fail closed).
```

Comment:
Start the second sentence with something like "Valid options are `Ignore` or `Fail`". I don't know exactly what the allow/fail open and block/fail closed parts mean, but I would try to be clearer if you can.

Reply:
I have now made this much clearer.